### PR TITLE
Update Az registration script - Support for Ruggedized billing model, re-registration, CustomCloud initialization

### PR DESCRIPTION
**Pull request by rakku-ms**

### CODE REVIEW

File: Registration/RegisterWithAzure.psm1 [hunk 1]
```diff
@@ -9,6 +9,45 @@ You must also have access to an account / directory that is an owner or contribu
...
+        [String] $CloudARMEndpoint
+    )
+
+    if ( $PSCmdlet.ParameterSetName -eq 'CloudManifest' ){
+        $cloudJson = Get-Content $CloudManifestFilePath -Raw | ConvertFrom-Json
+        $cloudJsonContent = $cloudJson.DeploymentData
+        $CloudARMEndpoint = $cloudJsonContent.CustomEnvironmentEndpoints.CustomCloudARMEndpoint
+    }
+
+    $fullUri = $CloudARMEndpoint.TrimEnd('/')+"/metadata/endpoints?api-version=2015-01-01"
```
Comment (bganapa) [inline Registration/RegisterWithAzure.psm1:31]: You need not query the metadata endpoint, instead use -ArmEndpoint

Comment (rakku-ms) [reply Registration/RegisterWithAzure.psm1:31]: I remember -ArmEndpoint method was giving error for EX cloud with some like unknown 'ResourceManagerEndpoint'. Also, can we set 'EnableAdfsAuthentication = $true' with -ARMEndpoint method?

Comment (bganapa) [reply Registration/RegisterWithAzure.psm1:31]: Ah , i see. You cannot set EnableAdfsAuthentication with the -ArmEdnpoint parameter set, You can do it Set-AzEnvironment after doing Add

What you have is fine if we need to set EnableAdfsAuthentication

**Comment by bganapa** [comment]
@rakku-ms please update the title of the PR with few details

**Pull request merged**